### PR TITLE
92 remove additional dependencies on digital land frontend

### PR DESCRIPTION
**Pull request by GeorgeGoodall**
Ticket: [https://trello.com/b/EnblBHSU/dluhc-data-team-delivery](https://trello.com/b/EnblBHSU/dluhc-data-team-delivery)

Removes the scss and js dependencies from dl-frontend

### CODE REVIEW

File: application/core/filters.py [hunk 1]
```diff
@@ -206,3 +211,54 @@ def cacheBust(uri):
...
+
+def commanum_filter(v):
+    """
+    Makes large numbers readable by adding commas
+
+    E.g. 1000000 -> 1,000,000
+    """
+    if isinstance(v, numbers.Number):
+        return "{:,}".format(v)
+    return v
```
Comment (GeorgeGoodall) [inline application/core/filters.py:264]: adding these filters from dl-frontend directly into the codebase

File: application/templates/entity.html [hunk 2]
```diff
@@ -265,13 +265,6 @@ <h3 class="govuk-heading-s">{{ type|capitalize }}</h3>
     </script>
   {% endif %}
 
-  <script>
-    const $subNavTabs = document.querySelector('[data-module="dl-subnav"]')
-    const subNavTabsComponent = new DLFrontend
-      .SubNavTabs($subNavTabs)
-      .init({})
-  </script>
-
```
Comment (GeorgeGoodall) [inline application/templates/entity.html:274]: document.querySelector('[data-module="dl-subnav"]') finds nothing. so this code emits an error. removing

File: assets/stylesheets/application.scss [hunk 3]
```diff
@@ -4,9 +4,8 @@ $govuk-assets-path: "/static/govuk/assets/";
 $govuk-new-link-styles: true;
 $govuk-link-underline-offset: .2em;
 
-// // import all govuk frontend scss
-// ideally we would remove this dependence on frontend
-@import "node_modules/digital-land-frontend/digital-land-frontend/dl-frontend";
+// import all govuk frontend scss
+@import "node_modules/govuk-frontend/govuk/all";
```
Comment (GeorgeGoodall) [inline assets/stylesheets/application.scss:8]: previously, govuk scss was imported via dl-frontend. we should just directly import it instead

File: assets/stylesheets/application.scss [hunk 4]
```diff
@@ -39,6 +38,13 @@ $govuk-link-underline-offset: .2em;
 @import "component/sub-navigation";
 @import "component/service-alert";
 
+@import "component/map/map";
+@import "component/header-bar";
+@import "component/list-filter/list-filter";
+
+@import "utilities/wrapper";
+@import "utilities/js-hidden";
+
```
Comment (GeorgeGoodall) [inline assets/stylesheets/application.scss:47]: we were using some scss from the dl-land frontend. I've just pulled those files in and imported them here. 
as a future ticket. I think all the scss could be refined. but in the interest of not making this PR too big, that's for another ticket

File: assets/stylesheets/component/_filter-group.scss [hunk 5]
```diff
@@ -22,3 +22,94 @@
...
+.dl-filter-group__auto-filter__input {
+  padding-left: 30px;
+  // sass-lint:disable indentation
+  background:
+    url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 36 36' width='40' height='40'%3E%3Cpath d='M25.7 24.8L21.9 21c.7-1 1.1-2.2 1.1-3.5 0-3.6-2.9-6.5-6.5-6.5S10 13.9 10 17.5s2.9 6.5 6.5 6.5c1.6 0 3-.6 4.1-1.5l3.7 3.7 1.4-1.4zM12 17.5c0-2.5 2-4.5 4.5-4.5s4.5 2 4.5 4.5-2 4.5-4.5 4.5-4.5-2-4.5-4.5z' fill='currentColor'%3E%3C/path%3E%3C/svg%3E")
+    $govuk-body-background-colour
+    no-repeat
+    -5px -3px;
+  // sass-lint:enable indentation
+}
```
Comment (GeorgeGoodall) [inline assets/stylesheets/component/_filter-group.scss:115]: dl-filter-group scss taken from dl-frontend

File: application/templates/entity.html (hunk 2 above)
Comment (mdsmith-scottlogic) [reply application/templates/entity.html:274]: What do you think was supposed to happen? Any history on when the subNavTabs element was removed?

File: default.paths.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  "templatesPath": "'application/templates'",
+  "staticFilesPath": "'application/static'",
+  "watchPaths": "'./src/scss/**/*.scss' './src/javascripts/**/*.js'",
+  "stylesheetsOutputPath": "assets/stylesheets",
+  "jsOutputPath": "assets/javascripts",
+  "imagesOutputPath": "assets/images",
+  "govukOutputPath": "assets/",
+  "rollupConfig": "./rollup.config.js",
+  "serverURL": "http://localhost:5000"
+}
```
Comment (GeorgeGoodall) [inline default.paths.json:14]: configuration object used by nps taken from dl-frontend. imagine there's redundant config in here. but I think that should be picked up as part of another ticket

File: package.json [hunk 7]
```diff
@@ -31,6 +28,9 @@
   },
   "homepage": "https://github.com/digital-land/digital-land.info#readme",
   "dependencies": {
-    "digital-land-frontend": "https://gitpkg.now.sh/digital-land/digital-land-frontend/package?main"
+    "copyfiles": "^2.4.1",
+    "govuk-frontend": "4.2.0",
+    "node-sass": "^9.0.0",
+    "nps": "^5.10.0"
```
Comment (GeorgeGoodall) [inline package.json:34]: removed the dependency on dl-frontend. but we still want some of the packages that dl-frontend was importing

File: application/templates/entity.html (hunk 2 above)
Comment (GeorgeGoodall) [reply application/templates/entity.html:274]: so here is the commit that removed the sub nav:
commit: https://github.com/digital-land/digital-land.info/commit/d797a5d1a0a02f2e75a6d7e0c89beef0ace88afd
and here is the PR that it said commit was part of:
PR: https://github.com/digital-land/digital-land.info/pull/48

my guess from the PR is that the entity page just underwent some large changes and its just a bit of code accidently left in from then

**Review by mdsmith-scottlogic — APPROVED** [review]
LGTM

**Pull request merged**